extensions/oidc-db-token-state-manager/runtime/src/main/java/io/quarkus/oidc/db/token/state/manager/runtime/OidcDbTokenStateManagerInitializer.java

```diff
@@ -0,0 +1,145 @@
+package io.quarkus.oidc.db.token.state.manager.runtime;
+
+import static io.quarkus.oidc.db.token.state.manager.runtime.OidcDbTokenStateManager.now;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import jakarta.enterprise.event.Observes;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Pool;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+
+public class OidcDbTokenStateManagerInitializer {
+
+    private static final Logger LOG = Logger.getLogger(OidcDbTokenStateManagerInitializer.class);
+    private static final String FAILED_TO_CREATE_DB_TABLE = "unknown reason, please report the issue and create table manually";
+    /**
+     * Extra 30 seconds before we delete expired tokens.
+     */
+    private static final long EXPIRED_EXTRA_GRACE = 30;
+
+    void initialize(@Observes StartupEvent event, OidcDbTokenStateManagerRunTimeConfig config, Vertx vertx, Pool pool,
+            OidcDbTokenStateManagerInitializerProperties initializerProps) {
+        if (config.createDatabaseTableIfNotExists()) {
+            createDatabaseTable(pool, initializerProps.createTableDdl, initializerProps.supportsIfTableNotExists);
+        }
+        periodicallyDeleteExpiredTokens(vertx, pool, config.deleteExpiredDelay().toMillis());
+    }
+
+    private static void periodicallyDeleteExpiredTokens(Vertx vertx, Pool pool, long delayBetweenChecks) {
+        vertx
+                .setPeriodic(5000, delayBetweenChecks, new Handler<Long>() {
```

Comment:
`setPeriodic` returns an id that you could store in a field.

Then, in a method that observes the `ShutdownEvent`, invoke `vertx.cancelTimer(id)`.

Reply:
Hey @tsegismont thanks for the comments, apologies I did not wait enough. As far as I recall, the timer is started on the Vert.x underlying HTTP router and therefore it will be cancelled as part of its closing sequence. I have a few of them in the OIDC code. 
Michal, if it proves to be a problem then we can handle all of them as it won't be a problem related to your PR only, please open an issue to investigate. I have another timer in the UserInfo/TokenIntrospection cache, so with this one - 3 in total, though as far as I recall those 2 I created they are activated optionally

Reply:
>I use injected Vertx instance, therefore I thought when Vert.X is closed on shutdown, the periodic is going to be closed as well. 

I thought the same, I might've heard from someone else though.

Reply:
@tsegismont If you'd like we can reopen this PR or indeed follow with the necessary cleanup as Michal proposed if necessary, thanks

Reply:
@sberyozkin I suggest to wait things are cleared out before doing anything. Also I think it will be much clearer to handle it in a separate PR for changes will be easier to review.

Reply:
> hey @tsegismont , thanks for taking your time, I'll come with PR that address your comment.
> 
> Truth be told, I had this in one of my iterations, but I had 2 reasons to drop it:
> 
>     * I use injected Vertx instance, therefore I thought when Vert.X is closed on shutdown, the periodic is going to be closed as well. When I look into implementation, I don't see it though.

When a Vert.x instance is closed, all timers are indirectly stopped by calling `EventExecutorGroup#shutdownGracefully`:

https://github.com/eclipse-vertx/vert.x/blob/23eb20ffba301d735b227c604a67e5c7ec1de234/src/main/java/io/vertx/core/impl/VertxImpl.java#L839

> 
>     * I saw Sergey is already using it without cancelling in `io.quarkus.oidc.runtime.BackChannelLogoutTokenCache`, so I thought it's fine
> 
> 
> If I get it right, periodic is scheduled on `EventLoop` and if we don't cancel it, we could risk resource leaks, e.g. in DEV mode?

If you're absolutely sure the corresponding Vert.x instance is closed when this bean instance is destroyed, then you may omit to properly cancel the timer.

From my perspective it would be safer (and does not cost much) to do it explicitly.

Reply:
> Hey @tsegismont thanks for the comments, apologies I did not wait enough

No worries, really.



> @sberyozkin I suggest to wait things are cleared out before doing anything. Also I think it will be much clearer to handle it in a separate PR for changes will be easier to review.

:+1: 

I don't believe it's necessary to revert the change and reopen the PR. I hope my comment above helps and don't hesitate to ping me if it doesn't.